src/imageLoader/wadouri/dataSetCacheManager.js

```diff
@@ -14,17 +15,90 @@ let loadedDataSets = {};
 
 let promises = {};
 
+function isMultiFrame(uri) {
+  const dataSet = _get(uri);
+
+  return _isMultiFrame(dataSet);
+}
+
+function _isMultiFrame(dataSet) {
+  // Checks if dicomTag NumberOf Frames exists and it is greater than one
+  if (!dataSet) {
+    return false;
+  }
+
+  const numberOfFrames = dataSet.intString('x00280008');
+
+  return numberOfFrames && numberOfFrames > 1;
+}
+
+function retrieveFrameParameterIndex(uri) {
+  return uri.indexOf('&frame=');
+}
+
+function _get(uri) {
+  if (!loadedDataSets[uri]) {
+    return;
+  }
+
+  return loadedDataSets[uri];
+}
+
+function retrieveFirstFrameMetadata(uri) {
+  const frameParameterIndex = retrieveFrameParameterIndex(uri);
+  const multiframeURI = uri.slice(0, frameParameterIndex);
+  const frame = parseInt(uri.slice(frameParameterIndex + 7), 10);
+
+  let dataSet;
+
+  if (loadedDataSets[uri]) {
+    dataSet = loadedDataSets[multiframeURI].dataSet;
+  } else {
+    dataSet = undefined;
+  }
+
+  return {
+    dataSet,
+    frame,
+  };
+}
+
```

Comment:
same here, don't pollut the dataset cache manager. Put them somewhere else and import them

Reply:
Done